xml/art_virtualization-best-practices.xml

```diff
@@ -3241,6 +3241,162 @@ LC_ALL=C PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
       </para>
 <screen>&prompt.root;<command>echo 1 > /sys/devices/system/cpu/cpu<replaceable>X</replaceable>/online</command></screen>
     </sect2>
+
+    <sect2 xml:id="sec-vm-guest-scsi-pr">
+      <title>SCSI3 persistent reservation (S3-PR) on a multipathed device</title>
+      <para>
+        SCSI persistent reservations allow restricting access to block devices
+        to specific initiators in a shared storage setup.  When implementing
+        clustering of &vmguest;, it is a common requirement for &vmguest; to
+        send persistent reservation SCSI commands.  However, the operating
```

Comment:
Why is it a "common requirement" for VM Guests to "send persistent reservation SCSI commands"? What are are these commands?

Reply:
Because it supports multiple nodes accessing a device while at the same time blocking access to other nodes, to prevent unexpected and dangerous concurrent access.

If you think the 'commands' is improper, I'll use the 'command' instead.

Reply:
Please refer to the example in this pr. I already mentioned it: "then perform following commands In VM..."


Start the &vmguest; then perform following commands In &vmguest;:

lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
sda      8:0    0   50G  0 disk
vda    253:0    0   20G  0 disk
├─vda1 253:1    0    8M  0 part
├─vda2 253:2    0    2G  0 part [SWAP]
└─vda3 253:3    0   18G  0 part /

Let's reserve the scsi disk with the key 123abc, then release it:
sg_persist --verbose --out --register --param-sark=123abc /dev/sda
...


So it's easy to know the scsi pr command is sent by the user/administrator of the guest OS, rather than sysadmin administering the VMs.
Of course, The details about how the scsi pr command is sent to the lun by qemu-pr-helper program, It isn't mentioned in this pr, and I don't think we need to show this details in this pr.